### ISSUE TITLE
Add missing accessibility support up to AA

### ISSUE THREAD

**Issue by peter-dick** [Infineon/infineon-design-system-stencil#1948]
### Detailed description of how to reproduce the problem

In https://github.com/Infineon/infineon-design-system-stencil/issues/1264 it is mentioned that all components support level AA accessibility.
In scope of IDCO evaluation, we discovered that some components do not fully support level A or AA.
The found issues:

* Components:
  - Navbar:
    -> Logo image link is missing text alternative (A)
    -> No keyboard navigation supported
  - Pagination:
    -> Buttons are missing a text alternative (A)
  - Select (Single):
    -> Add ability to add aria label
  - IconButton:
    -> AriaLabel not set on button / missing alternative text (A)
    -> The ifx-icon-button does not clearly provide its function (e.g. having role="button"), and the underlying button element does not have the mentioned aria-label set from ifx-icon-button.
  - Modal:
    -> Close button on the top right corner has no alternative text (A)
  - Multiselect/Select:
    -> Keyboard navigation not fully working (e.g. dropdown not closing on focus-lost/escape)
  - Dropdown:
    -> Keyboard navigation not possible (items not focusable, clickable)
* Color Contrast:
    Often contrast ration of 3.98:1 instead of 4.5:1 (e.g. placeholders like in select or multiselect)

### Upload a screenshot

No screenshot required

### Describe the development framework

Vue3 + DDS 35.2.0

### Expected behavior

All components can be controlled via keyboard, required contrast aspects for AA are respected, aria-labels or alt-texts can be set there required.

**Comment by peter-dick** [comment]
> Hi [@peter-dick](https://github.com/peter-dick) The topic of the accessibility is still ongoing. We are adding and releasing the necessary accessibility incrementally. All components will be enhanced with level AA accessibility. Some of the components you mentioned in this issue are currently under review, the rest are in the backlog to be implemented. I am therefore closing this issue for now. Feel free to open another one in case you have a specific request regarding an already implemented and released accessibility feature.

Thanks for the info @tishoyanchev. It would be great if you could provide us a list of issues / backlog items, so we can track when the respective items are completed or their status.

**Comment by tishoyanchev** [comment]
@peter-dick 
Unfortunately we don't have a list on our repository that we dynamically update, however I can send you a static list of the current status. Here you can see the components that are level AA  and completed on 100%, and the components that remain to be done. All components must be level AA. You can expect the list to be completed by the end of August. 

[wcag02_checklist.pdf](https://github.com/user-attachments/files/21410750/wcag02_checklist.pdf)
